### DISCUSSION TITLE
Sensor update to MQTT not happening with retained DeepSleepTime

### DISCUSSION THREAD

**Discussion by blacknell** [arendst/Tasmota#19685] [Q&A]
I've had this working fine on shorted DeepSleepTimes but it's going back to sleep before sending sensor data.

`TelePeriod 10` and `DeepSleepTime 300` work fine.

But when I extend the sleep period to an hour with `TelePeriod 10` and `DeepSleepTime 3600` I am finding that the MQTT sensor data is not being sent (or possibly recieved). Also Syslog logs are not being received either.

For example,

```
Oct  6 16:02:43 tasmota-CBF048-4168 ESP-HDW: ESP32-D0WD v1.0 
Oct  6 16:02:43 tasmota-CBF048-4168 ESP-UFS: FlashFS mounted with 12584 kB free
Oct  6 16:02:43 tasmota-CBF048-4168 ESP-CFG: Loaded from File, Count 1672
Oct  6 16:02:43 tasmota-CBF048-4168 ESP-CFG: CR 417/699, Busy 0
Oct  6 16:02:43 tasmota-CBF048-4168 ESP-SPI: Software using GPIO13(CLK), GPIO27(MOSI) and GPIO26(MISO)
Oct  6 16:02:43 tasmota-CBF048-4168 ESP-I2C: Bus1 using GPIO05(SCL) and GPIO25(SDA)

snip...

Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: MQTT
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 0, Cmd 'SETOPTION', Idx 52, Len 1, Pld 1, Data '1'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-MQT: stat/tasmota_CBF048/RESULT = {"SetOption52":"ON"}
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: MQTT
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 0, Cmd 'SETOPTION', Idx 19, Len 1, Pld 1, Data '1'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-MQT: stat/tasmota_CBF048/RESULT = {"SetOption19":"ON"}
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: MQTT
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 0, Cmd 'DEEPSLEEPTIME', Idx 1, Len 4, Pld 3600, Data '3600'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-MQT: stat/tasmota_CBF048/RESULT = {"DeepSleepTime":3600}
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: MQTT
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 1, Cmd 'BACKLOG', Idx 1, Len 55, Pld -99, Data 'Timezone 99; TimeStd 0,0,10,1,2,0; TimeDst 0,0,3,1,1,60'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: Backlog
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 0, Cmd 'TIMEZONE', Idx 1, Len 2, Pld 99, Data '99'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-MQT: stat/tasmota_CBF048/RESULT = {"Timezone":99}
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: MQTT
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 1, Cmd 'SETOPTION', Idx 53, Len 1, Pld 1, Data '1'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-MQT: stat/tasmota_CBF048/RESULT = {"SetOption53":"ON"}
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-SRC: Backlog
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-CMD: Grp 0, Cmd 'TIMESTD', Idx 1, Len 12, Pld 0, Data '0,0,10,1,2,0'
Oct  6 16:02:45 tasmota-CBF048-4168 ESP-MQT: stat/tasmota_CBF048/RESULT = {"TimeStd":{"Hemisphere":0,"Week":0,"Month":10,"Day":1,"Hour":2,"Offset":0}}
```
and that's the last log entry.

On MQTT Explorer I can see the `tasmota-CBF048/LWT` update at 16:02:45 as it goes to sleep but not `tasmota-CBF048/SENSOR` nor `tasmota-CBF048/STATE`

I think by publishing a retained sleep time of 3600 (or any value for that matter) the deepsleeptime command is being reexecuted every boot. Sometimes there's enough time for tasmota to send the SENSOR data but often not.

Could be as simple as documenting that a retained sleep time needs to be cleared on boot with publish2

Thoughts?

**Comment by barbudor** [comment]
Just wondering why you published a retained command as it by definition executed everytime Tasmota connects to the browser

The only case where I'm using a retained Deepsleeptime is with 0 to disable the deepsleep loop at next restart

We generally document what has to be done
If we were to document everything that has not to be done, we would explode github quota 😂

**Comment by blacknell** [comment]
Yes, I agree with you. I have logic running in HA which changes the interval based on time of day, sun set/rise, occupancy - so that’s the reason. I’ve tried a rule which clears the retained topic whenever deepsleeptime changes and that fixes my problem. RegardsPaulOn 6 Oct 2023, at 18:41, Barbudor ***@***.***> wrote:﻿
Just wondering why you published a retained command as it by definition executed everytime Tasmota connects to the browser
The only case where I'm using a retained Deepsleeptime is with 0 to disable the deepsleep loop at next restart
We generally document what has to be done
If we were to document everything that has not to be done, we would explode github quota 😂

**Reply by barbudor** [reply]
Ah that makes sense
You could also do that with rules in Tasmota
I've made similar rules, not based on time of the day but on battery level: https://tasmota.github.io/docs/DeepSleep/#rules-to-adapt-deepsleep-period-to-battery-level

Instead of using the batterty level to change the period, you could use the time from midnight
